src/test/java/upeldev/com/github/upel3/services/UserServiceTest.java

```diff
@@ -0,0 +1,182 @@
+package upeldev.com.github.upel3.services;
+
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.mockito.junit.jupiter.MockitoSettings;
+import org.mockito.quality.Strictness;
+import upeldev.com.github.upel3.model.Role;
+import upeldev.com.github.upel3.model.User;
+import upeldev.com.github.upel3.repositories.UserRepository;
+
+@ExtendWith(MockitoExtension.class)
+@MockitoSettings(strictness = Strictness.LENIENT)
+public class UserServiceTest {
+    @Mock
+    private UserRepository userRepository;
+
+    private UserService userService;
+
+    @BeforeEach
+    public void setup(){
+        userService = new UserService(userRepository);
+    }
+
+    @Test
+    @DisplayName("Should find users by their IDs")
+    public void findByIdTest(){
+        User user1 = new User("John", "Doe", "doe@gmail.com", "1234");
+        User user2 = new User("John", "Smith", "smith@gmail.com", "4321");
+
+        Long user1Id = user1.getId();
+        Long user2Id = user2.getId();
+        Long wrongId = 10L;
+
+        Mockito.when(userRepository.findUserById(user1Id))
+                .thenReturn(user1);
+        Mockito.when(userRepository.findUserById(user2Id))
+                .thenReturn(user2);
+        Mockito.when(userRepository.findUserById(wrongId))
+                .thenReturn(null);
+
+        Assertions.assertEquals(userService.findById(user1Id), user1);
```

Comment:
I think it would be an improvement to use AssertJ notation instead of the old JUnit one. Here for example it would be something like Assertions.assertThat(testedObject).isEqualTo(expectedObject). It's a bit better in terms of keeping the given/when/then convention.